cookbooks/aws-parallelcluster-slurm/files/default/head_node_slurm/slurm/templates/slurm_parallelcluster_queue_partition.conf

```diff
@@ -1,38 +1,89 @@
+{%- import 'common/slurm_utils.conf' as slurm_utils -%}
+
 # This file is automatically generated by pcluster
-{% set ns = namespace(has_node=false) %}
+{% set ns = namespace(has_node=false) +%}
 
-{% for compute_resource in queue_config.ComputeResources %}
-    {% set static_size = compute_resource.MinCount %}
-    {% set dynamic_size = compute_resource.MaxCount - compute_resource.MinCount %}
-    {% set instance_type = compute_resource.InstanceType%}
-    {% if static_size > 0 %}
-NodeName={{ queue_name }}-st-{{ compute_resource.Name }}-[1-{{ static_size }}] CPUs={{ compute_resource | vcpus }} RealMemory={{ compute_resource | realmemory }} State=CLOUD Feature=static,{{ instance_type }},{{ compute_resource.Name }}{% if compute_resource.Efa.Enabled %},efa{% endif %}{% if instance_type | gpus > 0 %},gpu{% if not no_gpu %} Gres=gpu:{{ instance_type | gpu_type }}:{{ instance_type | gpus }}{% endif %}{% endif %}
+{#- Macros: #}
+{%- macro cpus(compute_resource) -%}
+    {{ ' ' }}CPUs={{ compute_resource | vcpus }}
+{%- endmacro %}
 
-    {% endif %}
-    {% if dynamic_size > 0 %}
-NodeName={{ queue_name }}-dy-{{ compute_resource.Name }}-[1-{{ dynamic_size }}] CPUs={{ compute_resource | vcpus }} RealMemory={{ compute_resource | realmemory }} State=CLOUD Feature=dynamic,{{ instance_type }},{{ compute_resource.Name }}{% if compute_resource.Efa.Enabled %},efa{% endif %}{% if instance_type | gpus > 0 %},gpu{% if not no_gpu %} Gres=gpu:{{ instance_type | gpu_type }}:{{ instance_type | gpus }}{% endif %}{% endif %}
+{%- macro node_state(state="CLOUD") -%}
+    {{ ' ' }}State={{ state }}
+{%- endmacro %}
+
+{%- macro real_memory(compute_resource) %}
+    {%- if not compute_resource.InstanceTypeList -%}
+        {{ ' ' }}RealMemory={{ compute_resource | realmemory }}
+    {%- endif -%}
+{%- endmacro %}
+
+{%- macro feature(compute_resource, is_static) -%}
+    {{ ' ' }}Feature={{ "static" if is_static else "dynamic" }}
+    {%- if not compute_resource.InstanceTypeList -%}
+        ,{{ compute_resource.InstanceType }}
+    {%- endif -%}
+    ,{{ compute_resource.Name }}
+    {%- if compute_resource.Efa.Enabled -%}
+        ,efa
+    {%- endif -%}
+    {%- if compute_resource | instance_type_from_compute_resource | gpus > 0 -%}
+        ,gpu
+    {%- endif -%}
+{%- endmacro %}
+
+{%- macro gres(compute_resource) -%}
+    {% set instance_type = compute_resource | instance_type_from_compute_resource %}
+    {%- if (compute_resource | instance_type_from_compute_resource | gpus > 0) and not no_gpu -%}
+        {{ ' ' }}Gres=gpu:{{ instance_type | gpu_type }}:{{ instance_type | gpus }}
+    {%- endif -%}
+{%- endmacro -%}
 
+{% macro node_definition(queue_name, compute_resource, static_size, dynamic_size, state="CLOUD") %}
+    {% if static_size > 0 %}
+        {{- slurm_utils.node_name_section(queue_name, compute_resource.Name, static_size, True) -}}
+        {{- cpus(compute_resource) -}}
```

Comment:
I would be consistent in the naming of these macro. For example

```
node_name_section
cpus_section
...
```

or alternatively use `_setting` or `_property` as suffix. What's the name convention that Slurm uses for these?